src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -50,35 +50,45 @@ GetTotalEnsembleDurations(const ServerSideStats& stats)
   EnsembleDurations result;
   for (const auto& model_stats : stats.composing_models_stat) {
     if (model_stats.second.composing_models_stat.empty()) {
-      // Success count covers all successful requests, cumulative time, queue
-      // time, compute, and cache
-      const uint64_t cnt = model_stats.second.success_count;
-      // Infer/exec counts cover compute time done in inference backends,
-      // not related to cache hit times
-      const uint64_t infer_cnt = model_stats.second.inference_count;
+      result.total_queue_time_avg_us += AverageDurationInUs(
+          model_stats.second.queue_time_ns, model_stats.second.queue_count);
+      const uint64_t compute_input_time_avg_us = AverageDurationInUs(
+          model_stats.second.compute_input_time_ns,
+          model_stats.second.compute_input_count);
+      const uint64_t compute_infer_time_avg_us = AverageDurationInUs(
+          model_stats.second.compute_infer_time_ns,
+          model_stats.second.compute_infer_count);
+      const uint64_t compute_output_time_avg_us = AverageDurationInUs(
+          model_stats.second.compute_output_time_ns,
+          model_stats.second.compute_output_count);
+      const uint64_t compute_time_avg_us = compute_input_time_avg_us +
```

Comment:
If input/infer/output counts are being separated out to account for potential discrepencies between them, shouldn't average compute be computed like this instead? (in all places)
```
compute_time_avg_us = AverageDurationInUs(
  model_stats.compute_input_time_ns + model_stats.compute_infer_time_ns + model_stats.compute_output_time_ns,
  model_stats.second.compute_input_count + model_stats.second.compute_infer_count + model_stats.second.compute_output_count
)
```

Reply:
I agree...

Reply:
Actually I think I misspoke here. The total `compute_time_avg_us` should still just be divided by any one of the compute counts (not the sum of all three):
```
# Assuming compute_input_count == compute_infer_count == compute_output_count
compute_time_avg_us = AverageDurationInUs(
  model_stats.compute_input_time_ns + model_stats.compute_infer_time_ns + model_stats.compute_output_time_ns,
  model_stats.second.compute_infer_count
)
```

The check to validate that `compute_input_cnt == compute_infer_cnt == compute_output_cnt` and the individual averages is more along the lines of what I was thinking about when I wrote that comment, which you just added. Sorry for the back and forth. CC @tanmayv25 to confirm

Reply:
Switched to new calculation 👍